### DISCUSSION TITLE
Style menu based on menuPlacement when it is set to auto

### DISCUSSION THREAD

**Discussion by erdbom** [JedWatson/react-select#5811] [Q&A] [answered]
This is similar to #4872, I don't think it was ever answered fully. I have my menuPlacement set to auto, but I want to add styles to either the top or bottom of the menu depending on which side of the select it is rendered on. Is there a way to determine if the menu will render on top or bottom when menuPlacement is set to auto?

**Comment by manjushsh** [comment, marked as answer]
Hi @erdbom, yes we can detect it using combination of `menuPlacement or state.selectProps.menuPlacement` and `placement` props. as shown in image below.

<img width="589" alt="image" src="https://github.com/JedWatson/react-select/assets/94426452/d33ba949-d123-4697-be5b-818cca94557f">


👉🏼  [Code Sandbox link](https://codesandbox.io/s/affectionate-wood-wrt236?file=/src/other-stuff.js:154-384)

Example code:  
```
menu: (provided, state) => ({
    ...provided,
    ...(state.selectProps.menuPlacement === "auto"
      ? state?.placement === "top"
        ? { outline: "2px solid red" }
        : { outline: "2px solid green" }
      : {})
  }),
```

**Reply by erdbom** [reply]
Perfect, thank you!
